halyard-web/src/main/java/com/netflix/spinnaker/halyard/controllers/v1/AccountController.java

```diff
@@ -59,4 +72,57 @@ Account account(
 
     return result;
   }
+
+  @RequestMapping(value = "/{accountName:.+}", method = RequestMethod.PUT)
+  void setAccount(
+      @PathVariable String deploymentName,
+      @PathVariable String providerName,
+      @PathVariable String accountName,
+      @RequestParam(required = false, defaultValue = "false") boolean validate,
+      @RequestBody Object rawAccount) {
+    NodeReference reference = new NodeReference()
+        .setDeployment(deploymentName)
+        .setProvider(providerName)
+        .setAccount(accountName);
+
+    Account account = objectMapper.convertValue(
+        rawAccount,
+        Providers.translateAccountType(providerName)
+    );
+
+    Runnable doUpdate = () -> accountService.setAccount(reference, account);
```

Comment:
Good call - for now they call different functions `getAccount` vs `setAccount` but I'll generify it once I see a chance.